controllers/operands/wasp.go

```diff
@@ -0,0 +1,306 @@
+package operands
+
+import (
+	"errors"
+	"fmt"
+	"maps"
+	"os"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	sdkapi "kubevirt.io/controller-lifecycle-operator-sdk/api"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/common"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+const (
+	// For simplicity matters, re-use the virt-handler SA. That will spare
+	// the HCO controller from deploying custom privileged SA with SCC.
+	waspServiceAccount = "kubevirt-handler"
+	// Set dry run mode for e2e testing purposes
+	waspDryRunAnnotation = "wasp.hyperconverged.io/dry-run"
+)
+
+var volumeMounts = []string{"opt", "sys", "etc", "run", "tmp"}
+
+func newWaspHandler(Client client.Client, Scheme *runtime.Scheme) Operand {
+	return &conditionalHandler{
+		operand: &genericOperand{
+			Client: Client,
+			Scheme: Scheme,
+			crType: "DaemonSet",
+			hooks:  &waspHooks{},
+		},
+		getCRWithName: func(hc *hcov1beta1.HyperConverged) client.Object {
+			return NewWaspWithNameOnly(hc)
+		},
+		shouldDeploy: shouldDeployWasp,
+	}
+}
+
+type waspHooks struct {
+	cache *appsv1.DaemonSet
+}
+
+func (h *waspHooks) getFullCr(hc *hcov1beta1.HyperConverged) (client.Object, error) {
+	if h.cache == nil {
+		h.cache = NewWasp(hc)
+	}
```

Comment:
If you're using cache, please also implement the `reset()` method; see here for example: https://github.com/kubevirt/hyperconverged-cluster-operator/blob/5eb460e178f7c349f483e3586ff3694ec69cdf03/controllers/operands/cdi.go#L59-L61

Reply:
@nunnatsa Thanks! done